File: modules/format/src/main/scala/be/wegenenverkeer/atomium/format/pub/Draft.scala
package be.wegenenverkeer.atomium.format.pub

sealed trait Draft {
  def value:String
}

object Draft {
  def apply(value:String): Draft = if (value == DraftYes.value) DraftYes else DraftNo
}
object DraftYes extends Draft {
  override def value = "yes"
}

object DraftNo extends Draft {

  override def value = "no"
}

File: modules/core/src/main/java/be/wegenenverkeer/atomium/format/pub/Draft.java
package be.wegenenverkeer.atomium.format.pub;

import com.fasterxml.jackson.annotation.JsonValue;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlEnumValue;

@XmlEnum

public enum Draft {

    @XmlEnumValue("yes")
    YES,
    @XmlEnumValue("no")
    NO;

    @JsonValue
    public String getValue() {
        return name().toLowerCase();
    }
}
